### ISSUE TITLE
关于代码中upstage部分有些问题，麻烦作者可以解答一下我的疑惑吗

### ISSUE THREAD

**Issue by qiaomuqiao** [greekinRoma/DATransNet#6]
就是作者在写此处的前向传播时，好像和整体网络图中的Upstage不是完全对应啊，我有两处疑问。一，代码中写的是将upsample的结果和经过self.sattn（包含conv和sigmiod操作）的结果相乘了，而您是图中为什么是将DATrans的和self.sattn（包含conv和sigmiod操作）的结果相乘了；二，图像中的concat操作输入应该有一个是上采样后的结果，但此处的代码中的c和d似乎都不是经过上采样操作后的结果。可能是我没能完全看懂此处的输入d、c、xin我看图中输入的颜色不就两个吗，希望作者可以解答一下我的疑惑，谢谢您
class UpBlock_attention(nn.Module):
    def __init__(self, in_channels, out_channels, nb_Conv, activation='ReLU'):
        super().__init__()
        self.up = nn.Upsample(scale_factor=2,mode='bilinear')
        self.nConvs = _make_nConv(in_channels, out_channels, nb_Conv, activation)
        self.sattn = nn.Sequential(nn.Conv2d(in_channels//2,in_channels//2,kernel_size=1),
                                   nn.Sigmoid())
    def forward(self,d,c,xin):
        d = self.up(d)
        d = self.sattn(xin)*d
        x = torch.cat([c, d], dim=1)
        x = self.nConvs(x)
        return x

**Comment by greekinRoma** [comment]
> 作者做的太好了！感谢您的热心回答

谢谢师兄，哈哈哈哈哈。
